indexer/ai/src/main/java/com/google/cloudsearch/ai/CustomSkill.java

```diff
@@ -0,0 +1,138 @@
+package com.google.cloudsearch.ai;
+
+import com.google.common.collect.Multimap;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CustomSkill implements AISkill{
+
+    private String aiSkillName = "";
```

Comment:
why aiSkillName and outputMappings part of AISkill itself ?
These will be common to all skills right
Use an abstract class BaseSkill which can implement all the common functionalities
https://blog.udemy.com/difference-between-abstract-class-and-interface/